buku

```diff
@@ -1492,132 +1545,131 @@ class BukuDb:
+                     '\n    + '.join(map(_count, clauses)) +
+                     f') AS score\n  FROM bookmarks WHERE score > 0 ORDER BY score DESC, {_order})')
         elif all_keywords:
-            if len(keywords) == 1 and keywords[0] == 'blank':
-                q0 = "SELECT * FROM bookmarks WHERE metadata = '' OR tags = ? "
-                qargs += (DELIM,)
-            elif len(keywords) == 1 and keywords[0] == 'immutable':
-                q0 = 'SELECT * FROM bookmarks WHERE flags & 1 == 1 '
+            if keywords == ['blank']:
+                qargs, query = [DELIM], "SELECT * FROM bookmarks WHERE metadata = '' OR tags = ?"
+            elif keywords == ['immutable']:
+                qargs, query = [], 'SELECT * FROM bookmarks WHERE flags & 1 == 1'
             else:
-                q0 = 'SELECT id, url, metadata, tags, desc, flags FROM bookmarks WHERE '
-                for token in keywords:
-                    if not token:
-                        continue
-
-                    if deep:
-                        q0 += q1 + 'AND '
-                    else:
-                        _pre = _post = ''
-                        if str.isalnum(token[0]):
-                            _pre = '\\b'
-                        if str.isalnum(token[-1]):
-                            _post = '\\b'
-                        token = _pre + re.escape(token.rstrip('/')) + _post
-                        q0 += q2 + 'AND '
-
-                    qargs += (token, token, token, token,)
-
-                if not qargs:
-                    return []
-
-                q0 = q0[:-4]
-            q0 += 'ORDER BY id ASC'
+                query = 'SELECT id, url, metadata, tags, desc, flags FROM bookmarks WHERE ' + '\n  AND '.join(clauses)
+            query += f'\nORDER BY {_order}'
         elif not all_keywords:
-            q0 = 'SELECT id, url, metadata, tags, desc, flags FROM (SELECT *, '
-            for token in keywords:
-                if not token:
-                    continue
-
-                if deep:
-                    q0 += case_statement(q1) + ' + '
-                else:
-                    _pre = _post = ''
-                    if str.isalnum(token[0]):
-                        _pre = '\\b'
-                    if str.isalnum(token[-1]):
-                        _post = '\\b'
-                    token = _pre + re.escape(token.rstrip('/')) + _post
-                    q0 += case_statement(q2) + ' + '
-
-                qargs += (token, token, token, token,)
-
-            if not qargs:
-                return []
-
-            q0 = q0[:-3] + ' AS score FROM bookmarks WHERE score > 0 ORDER BY score DESC)'
+            query = ('SELECT id, url, metadata, tags, desc, flags\nFROM (SELECT *, (' +
+                     '\n    + '.join(map(_count, clauses)) +
+                     f') AS score\n  FROM bookmarks WHERE score > 0 ORDER BY score DESC, {_order})')
```

Comment:
Note that since `score` is first clause of `ORDER BY`, the `order` value only affects search results that have the same "score".